extensions/azurecore/src/azureResource/azure-resource.d.ts

```diff
@@ -21,8 +21,33 @@ export namespace azureResource {
 		readonly treeItem: TreeItem;
 	}
 
-	export interface AzureResourceSubscription {
+	export interface AzureResource {
+		name: string;
+	}
+
+	export interface AzureSqlResource extends AzureResource {
+		loginName: string;
+	}
+
+	export interface AzureResourceSubscription extends AzureResource {
 		id: string;
 		name: string;
 	}
```

Comment:
name field can be removed from AzureResourceSubscription since it gets this field inherited from the base AzureResource interface